internal/gitmetadata.go

```diff
@@ -121,3 +122,22 @@ func GetTracerGitMetadataTags() map[string]string {
 
 	return res
 }
+
+// removeCredentials returns the passed url with potential credentials removed.
+// If the input string is not a valid URL, the string is returned as is.
+func removeCredentials(urlStr string) string {
+	if urlStr == "" {
+		return urlStr
+	}
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		// not an url, nothing to remove
+		return urlStr
```

Comment:
Ah I didn't consider this possibility.

I think the main use-case we want to cover here is automatic injection of a secret, like directly from the git command, by running:
```
export DD_GIT_REPOSITORY_URL="$(git config --get remote.origin.url)"
```

I would expect this to always be a valid URL and so credentials stripping will work fine in this case.

The case where we may have invalid URLs is when the URL is manually entered. One case we currently support and want to keep supporting is `DD_GIT_REPOSITORY_URL="github.com/DataDog/dd-trace-go"`. This is not a valid URL but the backend can use this value today.

I'm fine not trying to remove credentials when the url is user-provided and may be incorrect, because this is a harder problem. Example: the user could set an incorrect but valid URL with credentials that won't be detected, like the URL `https://token@github.com/` could be reported as `https://tokengithub.com/` by mistake: this would be a credential leak and we would have no way to detect it, as it is a valid URL.

Reply:
Agreed, if we want to support `DD_GIT_REPOSITORY_URL="github.com/DataDog/dd-trace-go"` we need to return the original URL.